### PR TITLE
mute and unmute community and community chats

### LINKED ISSUE #3582: Mute and unmute community chats
When a community is muted the community chats are not muted which means the user can community chat notifications

## Acceptance Criteria
- When community is set as muted all community chats inherit the mute status of the community
- Unmuting a community chat or all community chats does not essentially means the community is unmuted. In this case the community remains muted until unmuted

### PR DESCRIPTION
**Pull request by jo-mut**
This adds the ability for community chats to inherit the mute status of the community but also unmute community chats when the community is muted

Closes #3582

### CODE REVIEW

File: protocol/communities_messenger_test.go [hunk 1]
```diff
@@ -3256,6 +3256,89 @@ func (s *MessengerCommunitiesSuite) TestSetMutePropertyOnChatsByCategory() {
...
+	s.Require().Len(mr.Communities(), 1)
+	s.Require().Len(mr.Chats(), 1)
+
+	mr, err = s.alice.CreateCommunityChat(newCommunity.ID(), orgChat2)
+	s.Require().NoError(err)
+	s.Require().NotNil(mr)
+	s.Require().Len(mr.Communities(), 1)
+	s.Require().Len(mr.Chats(), 1)
+
+	muteDuration, _ := s.alice.MuteDuration(MuteFor15Min)
```
Comment (cammellos) [inline protocol/communities_messenger_test.go]: err checking

File: protocol/communities_messenger_test.go [hunk 2]
```diff
@@ -3256,6 +3256,89 @@ func (s *MessengerCommunitiesSuite) TestSetMutePropertyOnChatsByCategory() {
...
+		}
+	}
+
+	chat := s.alice.Chats()[0]
+
+	err = s.alice.UnmuteChat(chat.ID)
+	s.Require().NoError(err)
+	s.Require().False(chat.Muted)
+
+	_, err = s.alice.MuteAllCommunityChats(newCommunity.IDString(), false, MuteFor15Min)
```
Comment (cammellos) [inline protocol/communities_messenger_test.go]: I prefer to have either two functions `MuteAllCommunityChats` and `UnmuteAllCommunityChats`, or at least rename the function so it's clear that is not just muting, something like `SetMutedOnAllCommunityChats` or similar

**Review by cammellos — COMMENTED** [review]
@jo-mut looks good, the only thing I can't see, how does the community gets unmuted after a set period of time?

**Review by cammellos — COMMENTED** [review]
@jo-mut looks good, the only thing I can't see, how does the community gets unmuted after a set period of time?

**Review by cammellos — COMMENTED** [review]
@jo-mut looks good, the only thing I can't see, how does the community gets unmuted after a set period of time?

**Comment by jo-mut** [comment]
yeah looks like i have not addressed that part. I will update the code implementing that

**Comment by ibrkhalil** [comment]
Duplicate of #3549

**Comment by jo-mut** [comment]
@cammellos seems this pr https://github.com/status-im/status-go/pull/3549 adds mute duration to a muted community and unmutes communities after a set period of time. I think the best thing to do is to roll back some of these changes in my pr and only merge mute and unmute community chats when a community is muted/unmuted. 

What do you think?

**Comment by jo-mut** [comment]
@borismelnik this adds ability to mute/unmute communities chats when the community is muted. Only community mute duration is duplicated here which am okay reverting so PRs have difference

File: protocol/communities/community_categories.go [hunk 3]
```diff
@@ -22,6 +22,20 @@ func (o *Community) ChatsByCategoryID(categoryID string) []string {
 	return chatIDs
 }
 
+func (o *Community) CommunityChats() []string {
```
Comment (borismelnik) [inline protocol/communities/community_categories.go]: this maybe `CommunityChatIDs`?

**Comment by borismelnik** [comment]
@jo-mut I have several questions:
1. If we mute community - all chats will be muted. But what if I unmute some chat? Should be the community muted property reset?
2. I see that community muted till can be set without `muted` property. What expected here?

**Comment by jo-mut** [comment]
@borismelnik I created a separate function to add muted-till property. The functions updates mute-till after community has been muted. They are two separate functions but basically achieves the same thing with your code

Unmuting a chat does not change the mute status of the community. Even if all chats are unmuted the community mute status does changes unless the community is also unmuted

**Comment by borismelnik** [comment]
> @borismelnik I created a separate function to add muted-till property. The functions updates mute-till after community has been muted. They are two separate functions but basically achieves the same thing with your code
> 
> Unmuting a chat does not change the mute status of the community. Even if all chats are unmuted the community mute status does changes unless the community is also unmuted

So if community muted, but one channel was unmuted, that's not means that community unmuted?

File: appdatabase/migrations/sql/1687367525_.up.sql
Comment (borismelnik) [inline appdatabase/migrations/sql/1687367525_.up.sql:1]: is it okay? empty name of migration file?

Comment (borismelnik) [reply appdatabase/migrations/sql/1687367525_.up.sql:1]: looks like you call `make migration D=""`

File: protocol/chat.go [hunk 4]
```diff
@@ -57,7 +57,7 @@ const (
 	MuteFor1Week
 	MuteTillUnmuted
 	MuteTill1Min
-	Unmuted
+	UnMuted
```
Comment (borismelnik) [inline protocol/chat.go]: @ilmotta correct me here to `Unmuted` cos its one word :)

**Review by borismelnik — APPROVED** [review]
Looks good :) Just a several suggestions:
  - Remove empty migration file
  - Be sure that `UnMuted` is expected instead of `Unmuted` (I know its looks better, but logically `Unmuted` is correct)

PS: I will approve it cos I believe you will fix it 👍

**Pull request merged**